src/test/java/bio/terra/app/model/CloudRegionTest.java

```diff
@@ -5,63 +5,54 @@
 import static org.hamcrest.Matchers.equalTo;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
-import bio.terra.common.category.Unit;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.exc.InvalidFormatException;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
-@AutoConfigureMockMvc
 @ActiveProfiles({"google", "unittest"})
-@Category(Unit.class)
+@Tag("bio.terra.common.category.Unit")
 public class CloudRegionTest {
 
-  @Autowired private ObjectMapper objectMapper;
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
```

Comment:
I wasn't sure if the Spring test overhead to get the autowired `ObjectMapper` bean was worth it here, so I converted this test case into a true unit test suite -- no Spring, no mocking.  Open to changing it back.

Reply:
IMO it's OK (and preferable) for a test to make its own object mapper unless what you're testing is the behavior of the injectable object mapper.

In this case what's being tested is `CloudRegionDeserializer` which is configured separately from the application object mapper (via annotations) so this usage seems OK to me. It's good be careful though, since an object mapper can configure which deserializers it knows about via an object mapper module, so the behavior might change depending on how the mapper that actually does the serialization is initialized.

I don't think it's needed here, but in the case where you are testing the bean provided object mappers you can also avoid injection by instantiating them directly e.g., by calling `ApplicationConfiguration.objectMapper()`.

Reply:
Thanks for engaging with me on this!  That's a great point… we're not testing the object mapper here.  I'm glad to know about the direct instantiation option for the future.